geopandas/tools/tests/test_sjoin.py

```diff
@@ -488,3 +506,189 @@ def test_sjoin_inner(self):
             self.cities, countries, how="inner", op="intersects"
         )
         assert cities_with_country.shape == (172, 4)
+
+
+@pytest.mark.skipif(
+    TEST_NEAREST,
+    reason=(
+        f"PyGEOS >= {MIN_PYGEOS_VERSION_FOR_NEAREST}"
+        " must be installed and activated via the geopandas.compat module to"
+        " test sjoin_nearest"
+    ),
+)
+def test_no_nearest_all():
+    df1 = geopandas.GeoDataFrame({"geometry": []})
+    df2 = geopandas.GeoDataFrame({"geometry": []})
+    with pytest.raises(
+        AttributeError, match="Expected the spatial index to implement `nearest_all`"
+    ):
+        sjoin_nearest(df1, df2)
+
+
+@pytest.mark.skipif(
+    not TEST_NEAREST,
+    reason=(
+        f"PyGEOS >= {MIN_PYGEOS_VERSION_FOR_NEAREST}"
+        " must be installed and activated via the geopandas.compat module to"
+        " test sjoin_nearest"
+    ),
+)
+class TestNearest:
+    @pytest.mark.skipif(TEST_NEAREST)
+    def setup_method(self):
+        df1_geoms = [Point(x, y) for x, y in zip(np.arange(2), np.arange(2))]
+        self.df1 = geopandas.GeoDataFrame({"geometry": df1_geoms})
+
+    @pytest.mark.parametrize("how", ("left", "right", "inner"))
+    @pytest.mark.parametrize("max_distance", (None, 1))
+    @pytest.mark.parametrize("distance_col", (None, "distance"))
+    def test_empty_right_df(self, how: str, max_distance: float, distance_col: str):
+        df2 = geopandas.GeoDataFrame({"geometry": []})
+        joined = sjoin_nearest(
+            self.df1, df2, how=how, max_distance=max_distance, distance_col=distance_col
+        )
+        if how != "left":
+            assert joined.empty
+            if distance_col is not None:
+                assert distance_col in joined
+        else:
+            assert (joined["geometry"] == self.df1["geometry"]).all()
```

Comment:
We have `assert_geoseries_equal` for this. Can you use that? Similarly in other places below.

Reply:
Yep, sorry for forgetting! Replaced.